### PR TITLE
Nix Build Expressions

### PR DESCRIPTION
**Pull request by expede**
Everything up to _but excluding_ Cachix. This PR adds a few commands so that building in pure Nix is possible:

```bash
get-nt # Install the Nix Tools
stack-to-nix # Make Nix derivations from stack.yaml

# Examples of building specific components via Nix
nix-build -A fission-core.components.library
nix-build -A fission-cli.components.exes.fission
```

Screen Shot 2021-04-05 at 10 24 42 PM

> haskell-nix [?] took 25m16s

25 minutes from scratch on my M1, so not _amazing_ but not _horrendous_... and now ready for Cachix

# TODO

* [ ] I'm actually going to clean up the commands a bit here first

### CODE REVIEW

**Comment by expede** [comment]
Screen Shot 2021-04-06 at 9 44 20 PM

**Comment by expede** [comment]
🙈 I think the Cachix action is building GHC from source because we need to add the IOHK substitutor

**Review by walkah — COMMENTED** [review]
Hrm. probably related to the cachix usage... testing this locally and `nix-build -A fission-cli.components.exes.fission` has been running for ... multiple hours ? It does seem to be compiling ghc from source (also appears to be using 8.8.4?)

File: nix-tools [hunk 1]
```diff
@@ -0,0 +1 @@
+/nix/store/4akrhslfa6w2fr6c8mg8lsli9v5fvbqb-nix-tools
```
[the remark this answers is not in the document]
Comment (expede) [reply nix-tools:1]: I mean, it doesn't _hurt_ but it's not _useful_. Will gitignore 👍

[the remark this answers is not in the document]
Comment (expede) [reply nix-tools:1]: > has been running for ... multiple hours ?

@walkah Yeah, that's building GHC from scratch then. You need to follow this: https://input-output-hk.github.io/haskell.nix/tutorials/getting-started/#setting-up-the-binary-cache

[the remark this answers is not in the document]
Comment (expede) [reply nix-tools:1]: (And rebuild switch, obvs)

**Comment by expede** [comment]
>  copying path '/nix/store/8777w3qhqd92h9y0d7lrcm8h5kvk7r2h-nix-tools-src' from 'https://hydra.iohk.io'...

Looking at least a bit better

**Comment by expede** [comment]
Screen Shot 2021-04-07 at 12 40 35 PM

Looks about right

**Pull request merged**